sections/tools/visual-regression-testing.md

```diff
@@ -0,0 +1,164 @@
+
+### Applitools flow
+
+Here is [the same test written with Applitools](https://github.com/muratkeremozcan/react-hooks-in-action-with-cypress/blob/main/cypress/e2e/ui-integration/user-context-retainment-applitools.spec.js).
+
+```javascript
+// Applitools version of the visual test
+// cypress/e2e/ui-integration/user-context-retainment-applitools.spec.js
+describe(
+  'User selection retainment between routes',
+  { tags: '@ui-integration' },
+  () => {
+    before(() => {
+      // Each test should open its own Eyes for its own snapshots
+      cy.eyesOpen({
+        appName: 'hooks-in-action',
+        testName: Cypress.currentTest.title
+      })
+
+      cy.stubNetwork()
+      cy.visit('/')
+    })
+
+    it('Should keep the user context between routes', () => {
+      cy.fixture('users').then((users) => {
+        cy.get('.user-picker').select(users[3].name)
+        cy.contains('Users').click()
+
+        cy.wait('@userStub')
+        cy.url().should('contain', '/users')
+        cy.get('.item-header').contains(users[3].name)
+
+        // full page test
+        cy.eyesCheckWindow({
+          tag: 'User selection retainment between routes',
+          target: 'window',
+          // if fully is true (default) then the snapshot is of the entire page,
+          // if fully is false then snapshot is of the viewport.
+          fully: false,
+          matchLevel: 'Layout'
+        })
+        // partial page test
+        cy.eyesCheckWindow({
+          tag: 'user details with custom selector',
+          target: 'region',
+          selector: '[data-cy="user-details"]'
+        })
+      })
+    })
+
+    afterEach(() => {
+      cy.eyesClose()
+    })
+  }
+)
+```
+
+We realize the additional `cy.eyesOpen` and  `cy.eyesClose` commands that need to execute in the beginning and end of the test. We also see ` cy.eyesCheckWindow` is very customizable, not needing a custom command as in Percy.
+
+> For details about setting up Applitools, and comparisons to Percy, check out [this blog post](https://dev.to/muratkeremozcan/setup-cypress-applitools-with-github-actions-a-comparison-of-applitools-vs-percy-in-a-mid-size-app-43ij).
+
+Similar to Percy, with Applitools our test executes cross browser & viewport, and records the base snapshot.
+
+![Image description](https://dev-to-uploads.s3.amazonaws.com/uploads/articles/uwxxha4w1wno1w076qwb.png)
+
+When / if there is a visual diff, there is a clear indicator in the web interface.
+
+![Image description](https://dev-to-uploads.s3.amazonaws.com/uploads/articles/wrrat01cd06zz134n1l9.png)
+
+Here is a side by side comparison of the Percy vs Applitools code.
```

Comment:
The comparison is lacking the custom `percySnapshotElement` custom command needed

Reply:
It's on L25 in user-context-retainment.spec.js, the left side.

Reply:
updated screenshot